lib/BloqadeLattices/src/tiles.jl

```diff
@@ -0,0 +1,39 @@
+abstract type Supercell{D} end
+
+struct Parallelepiped{D, T} <: Supercell{D}
+    # abstract matrix dimensions already guaranteed to be 2D
+    bounds_inv::AbstractMatrix{T}
+end
+
+struct Tile{D, T} <: Supercell{D}
+    bounds_inv::AbstractMatrix{T}
+end
+
+function Tile(bounds::AbstractMatrix{T}) where {T}
+    D = size(bounds,1)
+    @assert size(bounds,1) == size(bounds,2) # ensure proper square matrix
+
+    # return Tile
+    return Tile{D,T}(inv(bounds))
+end
+
+function Parallelepiped(bounds::AbstractMatrix{T}) where {T}
+    D = size(bounds,1)
+    @assert size(bounds,1) == size(bounds,2) # ensure proper square matrix
+
+    # return Parallelpiped
+    return Parallelepiped{D,T}(inv(bounds))
+end
+
+# check if a point is in the tile
+# Tile{D,T} is just concrete type based off the Supercell{D}
+# NTuple gives has D elements of type T, representing point
+in_range(x) = 0 ≤ x < 1 ? true : false
+within_cell(tile::Supercell{D},x::NTuple{D,T}) where {D,T} = all(in_range.(tile.bounds_inv * collect(x)))
```

Comment:
```suggestion
within_cell(cell::Parallelepiped{D, T},x::NTuple{D,T}) where {D,T} = all(in_range.(cell.bounds_inv * collect(x)))
```

Only need to implement this for specialized types